### PR TITLE
feat: (kernel) bulk delete by filters

### PR DESCRIPTION
**Pull request by lordmallam**
Makes available bulk delete by filters

Usage
`/{model_name}/filtered-deleted/?{query_string}`

### CODE REVIEW

File: aether-kernel/aether/kernel/api/tests/test_views.py [hunk 1]
```diff
@@ -928,3 +930,22 @@ def test_submission_validate(self):
...
+        mapping = models.Mapping.objects.first()
+        filtered_count = models.Entity.objects.filter(mapping=mapping.pk).count()
+        url = f'{reverse("entity-filtered-delete")}?mapping={str(mapping.id)}'
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(
+            models.Entity.objects.count(),
+            (entity_count - filtered_count)
+        )
+
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/tests/test_views.py:948]: could you add a test with no query string and check that everything is deleted? :scream:

File: aether-kernel/aether/kernel/api/views.py [hunk 2]
```diff
@@ -53,7 +53,23 @@
 )
 
 
-class ProjectViewSet(MtViewSetMixin, viewsets.ModelViewSet):
+class FilteredMixin(object):
+    @action(detail=False, methods=['delete'], url_path='filtered-delete')
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/views.py]: Question, why not `bulk-delete`? :thinking:

**Comment by obdulia-losantos** [comment]
Do you know this library https://github.com/miki725/django-rest-framework-bulk?

It could be a starting point for all the bulk actions (sadly, it's not maintained)

File: aether-kernel/aether/kernel/api/views.py [hunk 3]
```diff
@@ -53,7 +53,23 @@
...
+    def filtered_delete(self, request, *args, **kwargs):
+        qs = super(FilteredMixin, self).get_queryset()
+        filter_class = self.filter_class
+        try:
+            filtered_projects = filter_class(self.request.GET, queryset=qs)
```
Comment (obdulia-losantos) [inline aether-kernel/aether/kernel/api/views.py]: stupid comment, have to tried to do directly `qs.delete()` the `get_queryset()` method should have already filtered the query set, I think so :thinking:

**Comment by obdulia-losantos** [comment]
Proposal, why don't you move this piece of code to `aether-django-sdk-library` so we can use it in the rest of modules?

File: aether-kernel/aether/kernel/api/views.py (hunk 2 above)
Comment (lordmallam) [reply aether-kernel/aether/kernel/api/views.py]: 🤷‍♂  Bulk delete in my opinion feels like you have to pass the objects (ids) to be deleted.

Comment (obdulia-losantos) [reply aether-kernel/aether/kernel/api/views.py]: right

File: aether-kernel/aether/kernel/api/views.py (hunk 3 above)
Comment (lordmallam) [reply aether-kernel/aether/kernel/api/views.py]: Nope, Filter isn't applied on `get_queryset()`

Comment (obdulia-losantos) [reply aether-kernel/aether/kernel/api/views.py]: That's weird...

**Comment by lordmallam** [comment]
@obdulia-losantos I will move this to the django-sdk... Hold on.. 😁

**Comment by obdulia-losantos** [comment]
:scream: :scream: :scream: :scream:

**Review by obdulia-losantos — APPROVED** [review]

**Pull request merged**